commit 2a517d91af918aecd9b603b7a09f696d774de3e6
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Sun Apr 24 17:22:57 2016 +0200

    non-linux: Apply RFC3542 obsoletion of RFC2292 macros
    
    	RFC2292 macros were obsoleted by RFC3542, and should not be exposed
    	any more. Notably since IPV6_PKTINFO has been reintroduced with a
    	completely different API.
    
    	* bits/in.h (IPV6_PKTINFO): Rename to IPV6_2292PKTINFO.
    	(IPV6_HOPOPTS): Rename to IPV6_2292HOPOPTS.
    	(IPV6_DSTOPTS): Rename to IPV6_2292DSTOPTS.
    	(IPV6_RTHDR): Rename to IPV6_2292RTHDR.
    	(IPV6_PKTOPTIONS): Rename to IPV6_2292PKTOPTIONS.
    	(IPV6_HOPLIMIT): Rename to IPV6_2292HOPLIMIT.
    	(IPV6_RECVPKTINFO): New macro.
    	(IPV6_PKTINFO): New macro.

commit d454fd21f4366fc529f83fdcd8a1113c5b49853e
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Sun Apr 24 19:24:36 2016 +0200

    non-linux: Apply RFC3542 obsoletion of RFC2292 macros
    
    (IPV6_RECVHOPLIMIT, IPV6_HOPLIMIT, IPV6_RECVHOPOPTS, IPV6_HOPOPTS,
    IPV6_RTHDRDSTOPTS, IPV6_RECVRTHDR, IPV6_RTHDR, IPV6_RECVDSTOPTS,
    IPV6_DSTOPTS, IPV6_RECVPATHMTU, IPV6_PATHMTU, IPV6_DONTFRAG):
    New macros.

diff --git a/bits/in.h b/bits/in.h
index e6f4696..a9efc74 100644
--- a/bits/in.h
+++ b/bits/in.h
@@ -56,17 +56,17 @@ struct ip_opts
 
 /* IPV6 socket options.  */
 #define IPV6_ADDRFORM		1
-#define IPV6_PKTINFO		2
-#define IPV6_HOPOPTS		3
-#define IPV6_DSTOPTS		4
-#define IPV6_RTHDR		5
-#define IPV6_PKTOPTIONS		6
+#define IPV6_2292PKTINFO	2
+#define IPV6_2292HOPOPTS	3
+#define IPV6_2292DSTOPTS	4
+#define IPV6_2292RTHDR		5
+#define IPV6_2292PKTOPTIONS	6
 #define IPV6_CHECKSUM		7
-#define IPV6_HOPLIMIT		8
+#define IPV6_2292HOPLIMIT	8
 
-#define IPV6_RXINFO		IPV6_PKTINFO
-#define IPV6_TXINFO		IPV6_PKTINFO
-#define SCM_SRCINFO		IPV6_PKTINFO
+#define IPV6_RXINFO		IPV6_2292PKTINFO
+#define IPV6_TXINFO		IPV6_RXINFO
+#define SCM_SRCINFO		IPV6_TXINFO
 #define SCM_SRCRT		IPV6_RXSRCRT
 
 #define IPV6_UNICAST_HOPS	16
@@ -83,11 +83,15 @@ struct ip_opts
 #define IPV6_JOIN_ANYCAST      27
 #define IPV6_LEAVE_ANYCAST     28
 
+/* Advanced API (RFC3542) (1).  */
+#define IPV6_RECVPKTINFO	49
+#define IPV6_PKTINFO		50
+
 /* Obsolete synonyms for the above.  */
 #define IPV6_ADD_MEMBERSHIP	IPV6_JOIN_GROUP
 #define IPV6_DROP_MEMBERSHIP	IPV6_LEAVE_GROUP
-#define IPV6_RXHOPOPTS		IPV6_HOPOPTS
-#define IPV6_RXDSTOPTS		IPV6_DSTOPTS
+#define IPV6_RXHOPOPTS		IPV6_2292HOPOPTS
+#define IPV6_RXDSTOPTS		IPV6_2292DSTOPTS
 
 /* Routing header options for IPv6.  */
 #define IPV6_RTHDR_LOOSE	0	/* Hop doesn't need to be neighbour. */

diff --git a/bits/in.h b/bits/in.h
index a9efc74..d517115 100644
--- a/bits/in.h
+++ b/bits/in.h
@@ -86,6 +86,18 @@ struct ip_opts
 /* Advanced API (RFC3542) (1).  */
 #define IPV6_RECVPKTINFO	49
 #define IPV6_PKTINFO		50
+#define IPV6_RECVHOPLIMIT	51
+#define IPV6_HOPLIMIT		52
+#define IPV6_RECVHOPOPTS	53
+#define IPV6_HOPOPTS		54
+#define IPV6_RTHDRDSTOPTS	55
+#define IPV6_RECVRTHDR		56
+#define IPV6_RTHDR		57
+#define IPV6_RECVDSTOPTS	58
+#define IPV6_DSTOPTS		59
+#define IPV6_RECVPATHMTU	60
+#define IPV6_PATHMTU		61
+#define IPV6_DONTFRAG		62
 
 /* Obsolete synonyms for the above.  */
 #define IPV6_ADD_MEMBERSHIP	IPV6_JOIN_GROUP
